### PR TITLE
Introduce installation extras to lighten the package as dependency by default

### PR DESCRIPTION
**Pull request by asanin-epfl**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -11,13 +11,34 @@ Currently MorphTool provides:
...
+    pip install morph-tool[nrn]
+
+If plan to convert folders of morphologies:
+
+.. code:: bash
+
+    pip install morph-tool[parallel]
+
+
+
```
Comment (mgeplf) [inline README.rst:41]: mention 'all' here?

File: README.rst [hunk 2]
```diff
@@ -11,13 +11,34 @@ Currently MorphTool provides:
...
+    pip install morph-tool[plot]
+
+If plan to use `nrnhines` module:
+
+.. code:: bash
+
+    pip install morph-tool[nrn]
+
+If plan to convert folders of morphologies:
```
Comment (mgeplf) [inline README.rst:34]: does the folder converter work single process? or is dask a requirement?

**Review by mgeplf — COMMENTED** [review]
I think this is a nice change - I wonder how it will impact the spack install, or other consumers of the package, though?

File: README.rst (hunk 1 above)
Comment (asanin-epfl) [reply README.rst:47]: done

File: README.rst (hunk 2 above)
Comment (asanin-epfl) [reply README.rst:34]: `dask` is requirement but I wonder how it works because only `dask.bag` is used. I see no `Client` usage from `dask`.

Comment (arnaudon) [reply README.rst:34]: tbh, multiprocssing should work fine for that, if you want to remove dask dependency.

Comment (asanin-epfl) [reply README.rst:34]: don't step back from the righteous way of dask

Comment (arnaudon) [reply README.rst:34]: I wouldn't dare! haha

**Review by arnaudon — APPROVED** [review]

**Comment by wizmer** [comment]
Can you make sure that a meaningful error message is thrown if someone attemps to use a module without having the proper extra ?
A bit like https://github.com/BlueBrain/NeuroM/blob/master/neurom/view/plotly.py#L12 ?

**Comment by asanin-epfl** [comment]
I agree that explicit messages are required but now I think just to close this PR. I dropped `morph-tool` as dependency. Does anyone consider having this?

**Comment by arnaudon** [comment]
I don't mind having this or not at all. For me, this code does not have many depenencies I don't have elsewhere.

**Comment by wizmer** [comment]
I like this PR actually. I think it is a good thing, but you can drop it if you want.

**Review by wizmer — APPROVED** [review]

**Pull request merged**